Test failures with Python 3.13.0a4: test_dates_behave_like_dates and test_times_behave_like_times

The failures:

```
=================================== FAILURES ===================================
_________________________ test_dates_behave_like_dates _________________________

    def test_dates_behave_like_dates():
        i = item(date(2018, 7, 22))
    
        assert i == date(2018, 7, 22)
        assert i.as_string() == "2018-07-22"
    
        i += timedelta(days=1)
>       assert i == datetime(2018, 7, 23)
E       assert Date(2018, 7, 23) == datetime.datetime(2018, 7, 23, 0, 0)
E        +  where datetime.datetime(2018, 7, 23, 0, 0) = datetime(2018, 7, 23)

tests/test_items.py:692: AssertionError
_________________________ test_times_behave_like_times _________________________

    def test_times_behave_like_times():
        i = item(time(12, 34, 56))
    
        assert i == time(12, 34, 56)
        assert i.as_string() == "12:34:56"
    
>       i = i.replace(hour=13)

tests/test_items.py:726: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Time(12, 34, 56), args = (), kwargs = {'hour': 13}

    def replace(self, *args: Any, **kwargs: Any) -> time:
>       return self._new(super().replace(*args, **kwargs))
E       TypeError: Time.__init__() missing 2 required positional arguments: 'trivia' and 'raw'

tomlkit/items.py:1071: TypeError
=========================== short test summary info ============================
FAILED tests/test_items.py::test_dates_behave_like_dates - assert Date(2018, ...
FAILED tests/test_items.py::test_times_behave_like_times - TypeError: Time.__...
======================== 2 failed, 851 passed in 3.36s =========================
```

The changes around datetime and time in Python 3.13:
- time: https://github.com/python/cpython/pull/114780
- (possibly) date: https://github.com/python/cpython/pull/114760

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -17,7 +17,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-13, windows-latest]
-        python-version: [3.7, 3.8, 3.9, "3.10", "3.11", 3.12]
+        python-version: [3.7, 3.8, 3.9, "3.10", "3.11", 3.12, 3.13]
 
     steps:
       - uses: actions/checkout@v4
@@ -28,6 +28,7 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+          allow-prereleases: true
 
       - name: Get full python version
         id: full-python-version

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Change Log
 
+## [unreleased]
+
+### Fixed
+
+- Fix the incompatiblity with 3.13 because of the `datetime.replace()` change. ([#333](https://github.com/python-poetry/tomlkit/issues/333))
+
 ## [0.12.4] - 2024-05-08
 
 ### Fixed

--- a/tests/test_items.py
+++ b/tests/test_items.py
@@ -689,7 +689,7 @@ def test_dates_behave_like_dates():
     assert i.as_string() == "2018-07-22"
 
     i += timedelta(days=1)
-    assert i == datetime(2018, 7, 23)
+    assert i == date(2018, 7, 23)
     assert i.as_string() == "2018-07-23"
 
     i -= timedelta(days=2)
@@ -697,7 +697,7 @@ def test_dates_behave_like_dates():
     assert i.as_string() == "2018-07-21"
 
     i = i.replace(year=2019)
-    assert i == datetime(2019, 7, 21)
+    assert i == date(2019, 7, 21)
     assert i.as_string() == "2019-07-21"
 
     doc = parse("dt = 2018-07-22 # Comment")

--- a/tomlkit/items.py
+++ b/tomlkit/items.py
@@ -958,9 +958,14 @@ class Date(Item, date):
         return date.__new__(cls, year, month, day)
 
     def __init__(
-        self, year: int, month: int, day: int, trivia: Trivia, raw: str
+        self,
+        year: int,
+        month: int,
+        day: int,
+        trivia: Trivia | None = None,
+        raw: str = "",
     ) -> None:
-        super().__init__(trivia)
+        super().__init__(trivia or Trivia())
 
         self._raw = raw
 
@@ -1033,10 +1038,10 @@ class Time(Item, time):
         second: int,
         microsecond: int,
         tzinfo: tzinfo | None,
-        trivia: Trivia,
-        raw: str,
+        trivia: Trivia | None = None,
+        raw: str = "",
     ) -> None:
-        super().__init__(trivia)
+        super().__init__(trivia or Trivia())
 
         self._raw = raw
 

